src/DataValues/QuantityValue.php

```diff
@@ -473,14 +342,22 @@ public function getArrayValue() {
 	 * Constructs a new instance of the DataValue from the provided data.
 	 * This can round-trip with @see getArrayValue
 	 *
+	 * @note: if the upperBound or lowerBound field is missing from $data,
+	 * this returns an UnboundedQuantityValue, not a QuantityValue!
+	 *
 	 * @since 0.1
 	 *
 	 * @param mixed $data
 	 *
-	 * @return self
+	 * @return UnboundedQuantityValue
 	 * @throws IllegalValueException
 	 */
 	public static function newFromArray( $data ) {
+		if ( !isset( $data['upperBound'] ) || !isset( $data['lowerBound'] ) ) {
```

Comment:
Must be `&&`. As far as I remember I said this before: Please add tests for at least one possible combination to cover this.